### ISSUE TITLE
Tunnel to pod network missing with podCIDRStrategy=environment and podCIDRs defined

### ISSUE THREAD

**Issue by rkrzewski** [telepresenceio/telepresence#3478]
I am trying to connect to a new cluster using telepresence. 

The cluster is v1.28.5, installed using Charmed Kubernetes, running on bare metal machines & network. CNI backend is Calico running in BGP mode (no encapsulation).

Network setup is as follows: Kubernetes nodes are in `192.168.3.64/26` block, kubernetes API server IP (loadbalancer VIP) is in `192.168.0.64/26` and generally the `192.168.0.0/22` block is reserved for various infrastructural purposes.

Pod IP range in Calico config is set to `192.168.64.0/20` and service cluster IP block is `10.152.180.0/22`
I am trying to connect using Telepresence OSS v2.1.17. With the default configuration (after `telepresence helm install`) I am getting the following error:
```
telepresence connect: error: connector.Connect: subnet 192.168.0.0/17 overlaps with existing route "192.168.0.0/24 via 192.168.0.150 dev enp4s0, gw <nil>". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information
```
`192.168.0.0/24` is the network in my home LAN and `192.168.0.0/17` is obviously too broad. My understanding is that Telepresence is starting out with `nodePodCIDRs` strategy but not finding any (apparently Calico does not set `podCIDR` property on the nodes, and I've read that this property is deprecated and pending removal, but I can't find the reference just now). Then Telepresence tries `coverPodIPs` strategy but some of the pods no the cluster  (MetalLB speakers and such) have `hostNetwork: true` setting, and therefore their `podIP` is the IP of the node they're running at. Trying to cover both `192.168.3.64/26` (node) and `192.168.64.0/20` (pod) subnets at the same time is the likely reason for `192.168.0.0/17` subnet that swallows my LANs.

I have no need for the Telepresence tunnel to my k8s nodes because I can reach them via LAN/VPN but maybe it makes sense for other setups so I have no opinion on whether Telepresence should exclude `hostNetwork: true` pods when applying `coverPodIPs` strategy. This would likely resolve the problem I'm facing, but I am happy to override it by configuration for my particular setup. This does not work for me at the moment.

I've reinstalled the traffic manager with `telepresence helm upgrade --set podCIDRStrategy=environment --set podCIDRs=192.168.64.0/20`. I can see the following environement variables on `traffic-manager` pod:
```
      POD_CIDR_STRATEGY:            environment
      POD_CIDRS:                    192.168.64.0/20
```
I can see the following line in `traffic-manager` log
```
2024-01-08 20:09:19.4540 info    Using subnets from POD_CIDRS environment variable
```
However 192.168 network prefix does not appear further down in the log
In `daemon.log` I can see the following:
```
2024-01-08 21:20:09.6152 info    daemon/session/network : also-proxy subnets []
2024-01-08 21:20:09.6152 info    daemon/session/network : never-proxy subnets [192.168.3.5/32]
2024-01-08 21:20:09.6160 info    daemon/session/network : Creating routing table with index 775 and priority 32765
2024-01-08 21:20:09.6417 info    daemon/session/network : also-proxy subnets []
2024-01-08 21:20:09.6417 info    daemon/session/network : never-proxy subnets [192.168.3.5/32]
2024-01-08 21:20:09.6417 info    daemon/session/network : Adding Service subnet 10.152.180.0/22
2024-01-08 21:20:09.6417 info    daemon/session/network : Setting cluster DNS to 192.168.79.118
2024-01-08 21:20:09.6417 info    daemon/session/network : Setting cluster domain to "cluster.local."
```
Runing `ip route | grep tel` shows only the route to `10.152.180.0/22` network.

Notice that `daemon/session/network : Adding pod subnet` line is missing from `daemon.log`. I see such line when connecting to an old cluster running Telepresence v2.4.11, where I was also using `podCIDRStrategy=environment` setting. I am getting routes both to service and pod networks with the old cluster.

Please find the attached logs [telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/13866052/telepresence_logs.zip)

**Comment by thallgren** [comment]
Hey @rkrzewski . It seems like you're setting `--loglevel debug` from the CLI, which means that it's too late to see what's going on during connect.

Can you please try to instead declare it you config.yml file? E.g.
```yaml
logLevels:
  rootDaemon: debug
  userDaemon: debug
```

I suspect that the problem is that Telepresence detects that it already has pod connectivity (or connectivity to the pod subnets), so it doesn't add that subnet to the TUN-device.

**Comment by thallgren** [comment]
Can you also please provide the output of `telepresence config view`?

**Comment by rkrzewski** [comment]
Hi @thallgren, thanks for looking into it!

First off, helm chart config:
```
$ helm -n ambassador get values traffic-manager
USER-SUPPLIED VALUES:
agent:
  image:
    registry: docker.io/datawire
image:
  registry: docker.io/datawire
  tag: 2.17.0
podCIDRStrategy: environment
podCIDRs: 192.168.64.0/20
```
Configuration seen from the client:
```
$ telepresence config view
clientConfig:
    logLevels:
        userDaemon: debug
        rootDaemon: debug
clientFile: /home/rafal/.config/telepresence/config.yml
dns:
    remoteIP: 127.0.0.1
    excludeSuffixes:
        - .com
        - .io
        - .net
        - .org
        - .ru
    lookupTimeout: 8s
routing:
    subnets:
        - 10.152.180.0/22
    neverProxy:
        - 192.168.3.5/32
managerNamespace: ambassador
```
My machine has routes to a bunch of `192.168.n.n` subnets, but I don't think it has a route to `192.168.64.0/20`:
```
$ ip route | grep 192.168
default via 192.168.0.1 dev enp4s0 proto dhcp metric 100 
83.144.122.54 via 192.168.0.1 dev enp4s0 proto static metric 50 
192.168.0.0/24 dev enp4s0 proto kernel scope link src 192.168.0.150 metric 100 
192.168.0.1 dev enp4s0 proto static scope link metric 50 
192.168.2.0/24 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.3.0/24 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.12.0/24 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.122.134 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.123.0/24 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.200.0/22 via 10.13.38.41 dev tun0 proto static metric 50 
192.168.212.0/22 via 10.13.38.41 dev tun0 proto static metric 50
```
`tun0` is an OpenVPN interface. `enp4s0` is connected to a broadband provider's device.

I see the following in `daemon.log`:
```
2024-01-09 22:18:01.6933 debug   daemon/session/network : Performing pod connectivity check on IP 192.168.79.70 with timeout 500ms
2024-01-09 22:18:01.6934 debug   daemon/session/agentPods : WatchAgentPods starting
2024-01-09 22:18:02.1935 debug   daemon/session/network : Will proxy pods (context deadline exceeded)
2024-01-09 22:18:02.1936 debug   daemon/session/network : Performing service connectivity check on https://10.152.181.118:443/healthz with Host agent-injector.ambassador and timeout 500ms
2024-01-09 22:18:02.6937 debug   daemon/session/network : Will proxy services (Get "https://10.152.181.118:443/healthz": context deadline exceeded)
```
Which seems to indicate that telepresence should set up a tunnel for the pod subnet.

Today's logs: [telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/13879884/telepresence_logs.zip)

**Comment by thallgren** [comment]
Thanks. My suspicion was incorrect. I'll continue investigating this.

**Comment by thallgren** [comment]
Oh wait, this was fixed by PR #3472. That's why I can't reproduce it. I'm closing this ticket because the fix is in already and will be included in the next release.

**Comment by rkrzewski** [comment]
OK, thanks. I'll look out for the next release then.
